mozilla/fxa
# feat(react): Recreate oauth/signup functionality in React

## Because

* We're converting our legacy Backbone pages to React by extending fxa-settings

## This pull request

* Enable oAuth integration handling for the React Signup flow, including redirecting the user to the relying party after signup
* Update error handling for the ConfirmSignupCode page
* Update local storage utilities to allow updating the verification state of an account
* Create ConfirmSignupCode container component

## Issue that this pull request solves

Closes: #FXA-6519

## Checklist

_Put an `x` in the boxes that apply_


## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

### Testing
* Go to 123done and click “Email first”
* Append the experiment flow params, `&forceExperiment=generalizedReactApp&forceExperimentGroup=react`
* Verify the happy path takes you back to 123done
* Double check web flow (non-oauth) works as expected (CWTS / Sync still WIP)

Other things to look at (see code comments for why):
* Directly navigate to `/confirm_signup_code` in React (you’ll at least need the `showReactApp=true` param) and non-React version, you should be taken to Backbone `/signin` if in the OAuth flow for both cases and then Backbone `/confirm_signup_code`. For the web integration, it should load the React screen
* Try signing up for an unverified account then signing into 123done and opening a private window (or clear storage) and pasting the link & adding the experiment params before login attempt
